src/index.js

```diff
@@ -68,187 +88,39 @@ const initKalturaMux = function (player, options) {
-  // 1) Initial playback of the content via an autoplay mechanism
-  // 2) The user clicking play on the player
-  // 3) The user resuming playback of the video (by clicking play) after the player has been paused
-  // 4) Content playback is resuming after having been paused for an ad to be played inline (may require additional event tracking than the one below)
-  // player.on('playEvent', () => {
-  //   player.mux.emit('play');
-  // });
-
-  // Emit the `playing` event when the player begins actual playback of the content after the most recent
-  // `play` event. This should refer to when the first frame is displayed to the user (and when the next
-  // frame is presented for resuming from a paused state)
-  // player.on('playingEvent', () => {
-  //   player.mux.emit('playing');
-  // });
-
-  // NOTE: some players do not have an accurate `playing` event to use. In these scenarios, we typically track
-  // the first timeupdate with a playhead progression as the `playing` event, but send the event with a
-  // viewer_time back in time by the progressed amount. See below:
-  /*
-    player.on('playEvent'', () => {
-      const playTime = player.getCurrentTime();
-
-      // Listen for the first time update to be able to send PLAYING
-      let sendPlaying = (data) => {
-        const now = Date.now();
-        const currentTime = player.getCurrentTime();
-        const timeDiff = currentTime - playTime;
-
-        // Only send playing if we've progressed some
-        if (timeDiff > 0) {
-          // Unregister so it doesn't keep firing
-          player.off('timeupdateEvent', sendPlaying);
-          player.mux.emit('playing', {
-            viewer_time: now - secondsToMs(timeDiff)
-          });
-        }
+  player.ready().then(() => {
+    player.mux.emit('playerready', {});
+  });
+
+  PlaybackEventMap.forEach((kalturaEvent, muxEvent) => {
+    player.addEventListener(kalturaEvent, (event) => {
+      let data = {};
+
+      if (kalturaEvent === player.Event.Core.ERROR) {
+        data.player_error_code = event.payload.code;
+        data.player_error_message = event.payload.data.message;
       };
+      player.mux.emit(muxEvent, data);
+    });
+  });
 
-      player.on('timeupdateEvent', sendPlaying);
+  AdsEventMap.forEach((kalturaEvent, muxEvent) => {
+    player.addEventListener(kalturaEvent, (event) => {
+      let data = {};
 
-      // And clear this handler if we happen to get pause, error, seeking, or ended before timeupdate
-      player.on('pauseEvent', () => { player.off('timeupdateEvent', sendPlaying); });
-      player.on('endedEvent', () => { player.off(timeupdateEvent, sendPlaying); });
-      player.on('seekEvent', () => { player.off(timeupdateEvent, sendPlaying); });
-      player.on('errorEvent', () => { player.off(timeupdateEvent, sendPlaying); });
+      if (kalturaEvent === player.Event.AD_STARTED) {
+        const ad_tag_url = player.ads.getAd()._url;
+
+        player.mux.emit('ad_tag_url', ad_tag_url);
```

Comment:
`ad_tag_url` just needs to be assigned to `data` here, right?

Reply:
Oh awesome, so they do send us the asset url. Yeah what you're saying makes sense. We can emit ad_asset_url on adplaying, and ad_tag_url on ad_loaded.

Reply:
okay added the proper ad_tag_url and modified the ad_asset_url  fixed [here](2c07d959a59cafce3132ea53ad7036e242c8c853. 

<img width="1209" alt="Screen Shot 2021-10-14 at 4 19 05 PM" src="https://user-images.githubusercontent.com/90800572/137389683-7abebfa6-0582-4d0e-8ba8-9c7d2119b08b.png">
)